src/main/java/eu/stamp_project/mutationtest/descartes/operators/ThisMutationOperator.java

```diff
@@ -0,0 +1,57 @@
+package eu.stamp_project.mutationtest.descartes.operators;
+
+import org.pitest.classinfo.ClassName;
+import org.pitest.reloc.asm.MethodVisitor;
+import org.pitest.reloc.asm.Opcodes;
+import org.pitest.reloc.asm.Type;
+import org.pitest.reloc.asm.commons.Method;
+
+/**
+ * Replaces the method body with a <code>return this</code> statement
+ */
+public class ThisMutationOperator extends MutationOperator{
+
+    /**
+     * Returns a value indicating whether the operator can transform the given method.
+     * In this case whether <code>this</code> could be assigned to the return type.
+     *
+     * @param method Method to be tested by the operator
+     * @return A boolean value indicating if this can be assigned to the return type
+     */
+
+    @Override
+    public boolean canMutate(Method method) {
+        return true;
+    }
+
+    @Override
+    public boolean canMutate(ClassName className, Method method) {
+        int target = method.getReturnType().getSort();
+        String[] objArr = method.getReturnType().toString().split("/");
+        String[] classArr = className.asJavaName().split("\\.");
+        int check = objArr[objArr.length - 1].indexOf(classArr[classArr.length - 1]);
+        
+        if(check != -1){
+            return  target == Type.OBJECT || target == Type.ARRAY;
+        }
+        return true;
```

Comment:
This implementation is not correct. You only have to check if the return type of the method is the same as the class in which the method has been declared. If I understand correctly here you are only checking the last part of each name and not even for equality. The right check would be to compare both full names. `ClassName` has the `asInternalName` method. You need only to compare `className.asInternalName()` with `method.getReturnType().getInternalName()` or something similar, there are other alternatives.